src/SprykerEco/Zed/Unzer/Business/Reader/UnzerReader.php

```diff
@@ -0,0 +1,160 @@
+<?php
+
+/**
+ * MIT License
+ * For full license information, please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace SprykerEco\Zed\Unzer\Business\Reader;
+
+use Generated\Shared\Transfer\CustomerTransfer;
+use Generated\Shared\Transfer\PaymentUnzerOrderItemCollectionTransfer;
+use Generated\Shared\Transfer\PaymentUnzerOrderItemTransfer;
+use Generated\Shared\Transfer\PaymentUnzerTransfer;
+use Generated\Shared\Transfer\UnzerCredentialsCollectionTransfer;
+use Generated\Shared\Transfer\UnzerCredentialsConditionsTransfer;
+use Generated\Shared\Transfer\UnzerCredentialsCriteriaTransfer;
+use Generated\Shared\Transfer\UnzerCredentialsTransfer;
+use Generated\Shared\Transfer\UnzerCustomerTransfer;
+use SprykerEco\Zed\Unzer\Persistence\UnzerRepositoryInterface;
+
+class UnzerReader implements UnzerReaderInterface
+{
+    /**
+     * @var \SprykerEco\Zed\Unzer\Persistence\UnzerRepositoryInterface
+     */
+    protected $unzerRepository;
+
+    /**
+     * @var \SprykerEco\Zed\Unzer\Business\Reader\UnzerVaultReaderInterface
+     */
+    protected $unzerVaultReader;
+
+    /**
+     * @param \SprykerEco\Zed\Unzer\Persistence\UnzerRepositoryInterface $unzerRepository
+     * @param \SprykerEco\Zed\Unzer\Business\Reader\UnzerVaultReaderInterface $unzerVaultReader
+     */
+    public function __construct(
+        UnzerRepositoryInterface $unzerRepository,
+        UnzerVaultReaderInterface $unzerVaultReader
+    ) {
+        $this->unzerRepository = $unzerRepository;
+        $this->unzerVaultReader = $unzerVaultReader;
+    }
+
+    /**
+     * @param string $orderReference
+     *
+     * @return \Generated\Shared\Transfer\PaymentUnzerTransfer
+     */
+    public function getPaymentUnzerByOrderReference(string $orderReference): PaymentUnzerTransfer
+    {
+        return $this->unzerRepository->findPaymentUnzerByOrderReference($orderReference) ?? new PaymentUnzerTransfer();
+    }
+
+    /**
+     * @param string $orderId
+     *
+     * @return \Generated\Shared\Transfer\PaymentUnzerOrderItemCollectionTransfer
+     */
+    public function getPaymentUnzerOrderItemCollectionByOrderId(string $orderId): PaymentUnzerOrderItemCollectionTransfer
```

Comment:
the same as for repository
we could use only single method to retrieve collection of some entities by criteria
and consumer will grab 1st element from collection if needed

Reply:
will create tech depbt